lib/util/http/index.ts

```diff
@@ -82,15 +94,22 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
       url = URL.resolve(httpOptions.baseUrl, url);
     }
     // TODO: deep merge in order to merge headers
-    let options: any = {
+    let options: GotOptions = {
       method: 'get',
       ...this.options,
       hostType: this.hostType,
       ...httpOptions,
-    };
+    } as unknown; // TODO: fixme
     if (process.env.NODE_ENV === 'test') {
       options.retry = 0;
     }
+    // istanbul ignore if: only for easier testing
+    if (
+      ['GET'].includes(options.method?.toUpperCase()) &&
+      ('json' in options || 'body' in options)
+    ) {
+      throw new Error(`Method '${options.method}' does not support body`);
+    }
```

Comment:
I think this can be removed now, was to find wrong code. Maybe we need to fix some types to remove body for `GET` and `HEAD` requests.